workers/reconciliationworker.py

```diff
@@ -0,0 +1,151 @@
+import datetime
+import logging
+import time
+
+
+import features
+from app import app
+from app import (
+    billing as stripe,
+    rh_user_api as internal_user_api,
+    rh_marketplace_api as internal_marketplace_api,
+)
+from data import model
+from data.billing import get_plan, RH_SKUS
+from data.model import entitlements
+from util import marketplace
+from util.locking import GlobalLock, LockNotAcquiredException
+from workers.gunicorn_worker import GunicornWorker
+from workers.namespacegcworker import LOCK_TIMEOUT_PADDING
+from workers.worker import Worker
+
+
+logger = logging.getLogger(__name__)
+
+RECONCILIATION_TIMEOUT = 5 * 60  # 5min
+LOCK_TIMEOUT_PADDING = 60  # 60
+RECONCILIATION_FREQUENCY = 5 * 60  # run reconciliation every 5 min
+
+MILLISECONDS_IN_SECONDS = 1000
+SECONDS_IN_DAYS = 86400
+ONE_MONTH = 30 * SECONDS_IN_DAYS * MILLISECONDS_IN_SECONDS
+
+
+class ReconciliationWorker(Worker):
+    def __init__(self):
+        super(ReconciliationWorker, self).__init__()
+        self.add_operation(
+            self._reconcile_entitlements,
+            app.config.get("RECONCILIATION_FREQUENCY", RECONCILIATION_FREQUENCY),
+        )
+
+    def _check_stripe_matches_sku(self, user, sku):
+        """
+        Check if user's stripe plan matches with RH sku
+        """
+        stripe_id = user.stripe_id
+
+        if stripe_id is None:
+            return False
+
+        stripe_customer = stripe.Customer.retrieve(stripe_id)
+        if stripe_customer is None:
+            logger.debug("user %s has no valid subscription on stripe", user.username)
+            return False
+
+        plan = get_plan(stripe_customer.subscription.plan.id)
+        if plan is None:
+            return False
+
+        if plan.get("rh_sku") == sku:
+            return True
+
+        logger.debug(
+            "stripe plan %s and sku %s do not match for %s", plan["stripeId"], sku, user.username
+        )
+        return False
+
+    def _perform_reconciliation(self, user_api, marketplace_api):
+        """
+        Gather all entitlements from internal marketplace api and store in quay db
+        Create new entitlements for stripe customers if needed
+        """
+        logger.info("Reconciliation worker looking to create new subscriptions...")
+
+        users = model.user.get_active_users()
+
+        for user in users:
```

Comment:
Do you need to iterate every user, or only those with a stripe id?

Reply:
We can only look at stripe users, the only benefit I think going over all users is we can save account numbers for people that aren't using stripe with the reconciler